content/en/hosting/4.x/data-migration-3x-docker-to-4x-k3s-multi.md

```diff
@@ -0,0 +1,218 @@
+{{% /alert %}}
+
+## Migration Steps
+
+1. Install Migration Tool
+```shell
+mkdir -p ~/couchdb-migration/
+cd ~/couchdb-migration/
+curl -s -o ./docker-compose.yml https://raw.githubusercontent.com/medic/couchdb-migration/main/docker-compose.yml
+docker compose up
+```
+
+2. Set Up Environment Variables
+```shell
+# Replace with your actual CouchDB URL from the Docker Compose setup
+export COUCH_URL=http://<admin>:<password>@<couchdb-host>:5984
+```
+
+3. Run Pre-Migration Commands
+```shell
+cd ~/couchdb-migration/
+docker compose run couch-migration pre-index-views 4.10.0
+```
+
+{{% alert title="Note" %}} 
+If pre-indexing is omitted, 4.x API will fail to respond to requests until all views are indexed. For large databases, this could take many hours or days.
+{{% /alert %}}
+
+4. Save CouchDB Configuration
+```shell
+cd ~/couchdb-migration/
+docker compose run couch-migration get-env
+```
+
+Save the output containing:
+- CouchDB secret (used for encrypting passwords and session tokens)
+- CouchDB server UUID (used for replication checkpointing)
+- CouchDB admin credentials
+
+The next part of  the guide assumes your k3s cluster is already prepared. If not, please run the set of commands [here](https://docs.k3s.io/quick-start). Run k3s install command on the main node and run the k3s join commands on nodes 2 and 3.
+
+5. Prepare Node Storage
+
+```shell
+# Create directory on the primary node.
+# For production, make this a mounted disk instead of a directory on the root volume.
+sudo mkdir -p /srv/couchdb1/data
+
+# Copy data from Docker Compose installation
+sudo rsync -avz --progress --partial --partial-dir=/tmp/rsync-partial \
+    /srv/storage/medic-core/couchdb/data/ \
+    /srv/couchdb1/data/
+
+# Create directories on secondary nodes
+ssh user@node2-hostname "sudo mkdir -p /srv/couchdb2/data/shards /srv/couchdb2/data/.shards"
+ssh user@node3-hostname "sudo mkdir -p /srv/couchdb3/data/shards /srv/couchdb3/data/.shards"
+```
+
+6. Create values.yaml for K3s Deployment
+```yaml
+project_name: "your-namespace"
+namespace: "your-namespace"
+chtversion: 4.10.0
+
+upstream_servers:
+  docker_registry: "public.ecr.aws/medic"
+  builds_url: "https://staging.dev.medicmobile.org/_couch/builds_4"
+upgrade_service:
+  tag: 0.32
+
+couchdb:
+  password: "<your-saved-password>"
+  secret: "<your-saved-secret>"
+  user: "<your-admin-user>"
+  uuid: "<your-saved-uuid>"
+  clusteredCouch_enabled: true
+  couchdb_node_storage_size: 100Gi
+
+clusteredCouch:
+  noOfCouchDBNodes: 3
+
+ingress:
+  host: "your-url.org"
+
+environment: "remote"
+cluster_type: "k3s-k3d"
+cert_source: "specify-file-path"
+certificate_crt_file_path: "/home/henok/certs/fullchain.crt"
+certificate_key_file_path: "/home/henok/certs/privkey.key"
+
+nodes:
+  node-1: "couch01"
+  node-2: "couch02"
+  node-3: "couch03"
+
+couchdb_data:
+  preExistingDataAvailable: "true"
+  dataPathOnDiskForCouchDB: "data"
+  partition: "0"
```

Comment:
So here they're supposed to use the data storage that they just copied right? Is that mentioned somewhere?

Reply:
Is this done? I can't find a reference yet.

Reply:
Ah. Pushing an update now.

Reply:
Done.